### PR TITLE
Add documentation for object store layer.

### PR DESCRIPTION
**Pull request by OleksiiZubko**
Adding documentation with examples how to read and
write data to the object store layers, using functions
from `@here/olp-sdk-dataservice-api`

Resolves: OLPEDGE-2439

Signed-off-by: Oleksii Zubko <ext-oleksii.zubko@here.com>

### CODE REVIEW

File: docs/read-from-object-store-layer.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+   npm install --save @here/olp-sdk-authentication @here/olp-sdk-dataservice-api
+   ```
+
+   Now, everything is set to create the app.
+
+5. Create the `index.ts` file and app skeleton.
+
+   ```typescript
+   /**
+    * Example of the Node.js app used for reading/writing from/to object store layer from the datastore.
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:39]:
```suggestion
    * An example of the Node.js app used for reading data from and writing it to an object store layer in the datastore.
```

File: docs/read-from-object-store-layer.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+   tsc && node .
+   ```
+
+After a successful run, the console displays the following message:
+
+```shell
+App works!
+```
+
+## Use `ObjectStoreApi` from `@here/olp-sdk-dataservice-api`;
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:64]:
```suggestion
## Use `ObjectStoreApi` from `@here/olp-sdk-dataservice-api`
```

File: docs/read-from-object-store-layer.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+You can use the `ObjectStoreApi` functions to request any data from a [object store layer](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html#object-store-layers).
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:66]:
```suggestion
You can use the `ObjectStoreApi` functions to request any data from an [object store layer](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html#object-store-layers).
```

File: docs/read-from-object-store-layer.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+**To use the `ObjectStoreApi` functions:**
+
+1. Create the `OlpClientSettings` object.
+
+   For instructions, see <a href="https://github.com/heremaps/here-data-sdk-typescript/blob/master/docs/create-platform-client-settings.md" target="_blank">Create platform client settings</a>.
+
+2. Create a `RequestBuilder` instance with `RequestFactory` that contains the catalog HRN, the platform client settings from step 1, and api name `blob` and api version `v2`.
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:74]:
```suggestion
2. Create a `RequestBuilder` instance with `RequestFactory` that contains the catalog HRN, the platform client settings from step 1, API name, and API version.
```

File: docs/read-from-object-store-layer.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+   ```typescript
+   const requestBuilder = await RequestFactory.create(
+      "blob", 
+      "v2", 
+      settings, 
+      HRN.fromString("your-catalog-hrn")
+   );
+   ```
+
+## Get data from the object store layer
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:85]:
```suggestion
## Get data from an object store layer
```

File: docs/read-from-object-store-layer.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+You can request any data from a [object store layer](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html#object-store-layers) by key or by datahandle.
+
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:88]:
```suggestion
You can request any data from an [object store layer](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html#object-store-layers) using a key or datahandle.

```

File: docs/read-from-object-store-layer.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+**To get data from the object store layer:**
+
+1. Create the `RequestBuilder` object.
+
+   For instructions, see [Create RequestBuilder](#Use `ObjectStoreApi` from `@here/olp-sdk-dataservice-api`;).
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:93]:
```suggestion
   For instructions, see [Create RequestBuilder](#use-objectstoreapi-from-hereolp-sdk-dataservice-api).
```

File: docs/read-from-object-store-layer.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+   ```typescript
+   const result = await ObjectStoreApi.getBlobByKey(requestBuilder, {
+      key: "your-data-key",
+      layerId: "your-layer-id"
+   });
+   ```
+
+You receive data from the requested key of the selected layer.
+
+**To get all keys in the object store layer:**
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:106]:
```suggestion
## Get all keys

You can get all data keys from an [object store layer](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html#object-store-layers).

**To get all keys from the object store layer:**
```

File: docs/read-from-object-store-layer.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+1. Create the `RequestBuilder` object.
+
+   For instructions, see [Create RequestBuilder](#Use `ObjectStoreApi` from `@here/olp-sdk-dataservice-api`;).
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:110]:
```suggestion
   For instructions, see [Create RequestBuilder](#use-objectstoreapi-from-hereolp-sdk-dataservice-api).
```

File: docs/read-from-object-store-layer.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+   ```typescript
+   const result = await ObjectStoreApi.listKeys(requestBuilder, {
+      layerId: "your-layer-id"
+   });
+   ```
+
+You receive the response with list all keys in the selected layer.
+
+## Publish data to the object store layer
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:122]:
```suggestion
## Publish data to an object store layer
```

File: docs/read-from-object-store-layer.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+**To publish data to the object store layer:**
+
+1. Create the `RequestBuilder` object.
+
+   For instructions, see [Create RequestBuilder](#Use `ObjectStoreApi` from `@here/olp-sdk-dataservice-api`;).
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:128]:
```suggestion
   For instructions, see [Create RequestBuilder](#use-objectstoreapi-from-hereolp-sdk-dataservice-api).
```

File: docs/read-from-object-store-layer.md [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+2. Call the `putBlobByKey` function with the `key`, `data`, `contentLength` and `layerId` parameters.
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:130]:
```suggestion
2. Call the `putBlobByKey` function with the `key`, `data`, `contentLength`, and `layerId` parameters.
```

File: docs/read-from-object-store-layer.md [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+
+   ```typescript
+   const result = await ObjectStoreApi.putBlobByKey(requestBuilder, {
+      layerId: "your-layer-id",
+      key: "your-data-key",
+      contentLength: data.length,
+      body: data
+   });
+   ```
+You receive response from the datastore with the status of operation.
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:140]:
```suggestion
You receive a response from the datastore with the status of the operation.
```

File: docs/read-from-object-store-layer.md [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+
+For more information, you can visit the [Data User Guide](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html)
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:142]:
```suggestion
For more information, see the [Data User Guide](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html)
```

File: docs/read-from-object-store-layer.md [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+
+2. Call the `listKeys` function with the `layerId` parameter.
+
+   ```typescript
+   const result = await ObjectStoreApi.listKeys(requestBuilder, {
+      layerId: "your-layer-id"
+   });
+   ```
+
+You receive the response with list all keys in the selected layer.
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:120]:
```suggestion
You receive a list of all keys from the selected layer.
```

File: docs/read-from-object-store-layer.md (hunk 10 above)
Comment (HalynaDumych) [reply docs/read-from-object-store-layer.md:122]: Please apply this comment.

File: docs/read-from-object-store-layer.md [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+   const result = await ObjectStoreApi.listKeys(requestBuilder, {
+      layerId: "your-layer-id"
+   });
+   ```
+
+You receive a list of all keys from the selected layer.
+
+## Publish data to the object store layer
+
+**To publish data to an object store layer:**
```
Comment (HalynaDumych) [inline docs/read-from-object-store-layer.md:124]:
```suggestion
**To publish data to the object store layer:**
```

**Review by HalynaDumych — APPROVED** [review]

**Review by HalynaDumych — APPROVED** [review]

File: docs/read-from-object-store-layer.md [hunk 17]
```diff
@@ -0,0 +1,141 @@
+# Work with an object store layer
```
Comment (jkarjala) [inline docs/read-from-object-store-layer.md:1]: The file name "read-from-object-store-layer.md" is a bit misleading when this also covers write?

File: docs/read-from-object-store-layer.md [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+
+## Publish data to the object store layer
+
+**To publish data to the object store layer:**
+
+1. Create the `RequestBuilder` object.
+
+   For instructions, see [Create RequestBuilder](#use-objectstoreapi-from-hereolp-sdk-dataservice-api).
+
+2. Call the `putBlobByKey` function with the `key`, `data`, `contentLength`, and `layerId` parameters.
```
Comment (jkarjala) [inline docs/read-from-object-store-layer.md:129]: Will there be another doc entry for the multi-part upload wrapper?

File: docs/read-from-object-store-layer.md [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+   const result = await ObjectStoreApi.putBlobByKey(requestBuilder, {
+      layerId: "your-layer-id",
+      key: "your-data-key",
+      contentLength: data.length,
+      body: data
+   });
+   ```
+You receive a response from the datastore with the status of the operation.
+
+For more information, see the [Data User Guide](https://developer.here.com/documentation/data-user-guide/user_guide/portal/layers/layers.html)
```
Comment (jkarjala) [inline docs/read-from-object-store-layer.md:141]: Perhaps also the 'delete' case should be added?

File: docs/read-from-object-store-layer.md [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+   For instructions, see [Create RequestBuilder](#use-objectstoreapi-from-hereolp-sdk-dataservice-api).
+
+2. Call the `listKeys` function with the `layerId` parameter.
+
+   ```typescript
+   const result = await ObjectStoreApi.listKeys(requestBuilder, {
+      layerId: "your-layer-id"
+   });
+   ```
+You receive a list of all keys from the selected layer.
```
Comment (jkarjala) [inline docs/read-from-object-store-layer.md:119]: It should be noted that the listkeys returns only one level of keys. The response will contain for each key an indication if it is a folder or object. For each folder, the client must recursively call this end point with a parent in the query string to retrieve the next level of keys for the folder.

File: docs/read-from-object-store-layer.md (hunk 18 above)
Comment (OleksiiZubko) [reply docs/read-from-object-store-layer.md:129]: yes, after wrapper will be added

File: docs/read-from-object-store-layer.md (hunk 17 above)
Comment (OleksiiZubko) [reply docs/read-from-object-store-layer.md:1]: good catch, will rename. thanks

File: docs/using-object-store-layer.md [hunk 21]
```diff
@@ -0,0 +1,164 @@
...
+   const result = await ObjectStoreApi.listKeys(requestBuilder, {
+      layerId: "your-layer-id"
+   });
+   ```
+You receive a list of all keys from the selected layer for one level.
+The list can be like virtual directory-like structure created from keys in the 
+storage where '/' is used as a separator.
+The response will contain for each key an indication if it is a folder or object. For each folder, 
+the we must recursively call this function with a parent in the query string 
+to retrieve the next level of keys for the folder.
```
Comment (HalynaDumych) [inline docs/using-object-store-layer.md:124]:
```suggestion
You receive a list of all keys from one level of the selected layer.
The list is a virtual directory. Its structure consists of keys in
the storage where slashes (/) are used as separators.
For each key, the response indicates if it is a folder or object.
To retrieve the next level of keys in a folder,
call this function recursively with a parent in the query string.
```

**Review by HalynaDumych — CHANGES_REQUESTED** [review]

**Review by HalynaDumych — APPROVED** [review]

**Pull request merged**